src/libraries/illink-oob.targets

```diff
@@ -0,0 +1,67 @@
+
+    <PropertyGroup>
+      <LibrariesTrimmedOOBArtifactsPath>$([MSBuild]::NormalizePath('$(ArtifactsBinDir)', 'ILLinkTrimAssembly', '$(BuildSettings)', 'trimmed-oobs'))</LibrariesTrimmedOOBArtifactsPath>
+    </PropertyGroup>
+
+    <ItemGroup>
+
+      <!-- The following is the list of all the OOBs we will ignore for now -->
+      <_OOBsToIgnore Include="System.CodeDom" />
+      <_OOBsToIgnore Include="System.ComponentModel.Composition" />
+      <_OOBsToIgnore Include="System.ComponentModel.Composition.Registration" />
+      <_OOBsToIgnore Include="System.Composition.AttributedModel" />
+      <_OOBsToIgnore Include="System.Composition.Convention" />
+      <_OOBsToIgnore Include="System.Composition.Hosting" />
+      <_OOBsToIgnore Include="System.Composition.Runtime" />
+      <_OOBsToIgnore Include="System.Composition.TypedParts" />
+      <_OOBsToIgnore Include="System.Configuration.ConfigurationManager" />
+      <_OOBsToIgnore Include="System.Speech" />
+
+      <_OOBsToTrim Include="$(NetCoreAppCurrentRuntimePath)*.dll" Exclude="$(NetCoreAppCurrentRuntimePath)*.Generator.dll;$(NetCoreAppCurrentRuntimePath)*.Native.dll" />
+      <_RuntimePackTrimmedAssemblies Include="$(MicrosoftNetCoreAppRuntimePackRidLibTfmDir)*.dll" />
+
+      <!-- Move previous items to FileName so that we can subtract them -->
+      <_OOBsToFileName Include="@(_OOBsToTrim -> '%(FileName)')">
+        <OriginalIdentity>%(Identity)</OriginalIdentity>
+      </_OOBsToFileName>
+      <_RuntimePackAssembliesToFileName Include="@(_RuntimePackTrimmedAssemblies -> '%(FileName)')">
+        <OriginalIdentity>%(Identity)</OriginalIdentity>
+      </_RuntimePackAssembliesToFileName>
+
+      <_FinalOOBsFileName Include="@(_OOBsToFileName)" Exclude="@(_RuntimePackAssembliesToFileName);@(_OOBsToIgnore)" />
+      <_FinalOOBReferences Include="@(_OOBsToFileName)" Exclude="@(_FinalOOBsFileName)" />
+      <_FinalOOBs Include="@(_FinalOOBsFileName -> '%(OriginalIdentity)')" />
+      <_FinalReferences Include="@(_FinalOOBReferences -> '%(OriginalIdentity)')" />
+      <_FinalReferences Include="$(SystemPrivateCoreLibPath)" />
+
+      <OOBRootAssemblies Include="@(_FinalOOBs)">
+        <RootMode>library</RootMode>
+      </OOBRootAssemblies>
+    </ItemGroup>
+
+    <ItemGroup>
+      <!-- Include suppression XML files bin-placed in earlier per-library linker run. -->
+      <_OOBSuppressionsXmls Include="$(ILLinkTrimAssemblyOOBSuppressionsXmlsDir)*.xml" />
+    </ItemGroup>
+
+   <PropertyGroup>
+      <OOBILLinkArgs>$(ILLinkArgs)</OOBILLinkArgs>
+      <OOBILLinkArgs Condition="'@(_OOBSuppressionsXmls)' != ''" >$(OOBILLinkArgs) --link-attributes &quot;@(_OOBSuppressionsXmls->'%(FullPath)', '&quot; --link-attributes &quot;')&quot;</OOBILLinkArgs>
+    </PropertyGroup>
+
+    <ILLink AssemblyPaths=""
+        RootAssemblyNames="@(OOBRootAssemblies)"
+        ReferenceAssemblyPaths="@(_FinalReferences)"
```

Comment:
Passing these as ReferenceAssemblyPaths will set their action to skip - I think what you want is to pass them as AssemblyPaths, to get warnings from them when used as dependencies of the OOB libraries.

Reply:
IMO - this is intended behavior. When we are analyzing the OOB libraries, we don't want warnings from the runtimepack libraries.

But re-thinking about the whole scenario, if we are going to take this approach for analyzing the OOBs (and not the approach our customers would take), would it make sense to combine this all into 1 giant step, and analyze all the libraries (OOB and runtimepack) together?

Reply:
I was thinking of combining them as well, and to be honest I don't really have any objections if we want to do that approach as opposed to calling the linker twice. I went with the separate approach just because I originally understood that we didn't want to touch the way we are linking the shared framework today, but if your ok with it then that is great. The only downside I see on combining it, is that today the OOB linking step is not passing any noWarns per your comment suggestion and that is working, so if we were to combine them we would be linking the OOBs with the noWarns again. I'm happy to go either way here.

Reply:
> because I originally understood that we didn't want to touch the way we are linking the shared framework today, but if your ok with it then that is great.

My original concern with not affecting the shared framework analysis step was when we were going with the other approach. If we are taking this approach, I'm OK with affecting the shared fx analysis, where necessary.

> The only downside I see on combining it, is that today the OOB linking step is not passing any noWarns per your comment suggestion and that is working, so if we were to combine them we would be linking the OOBs with the noWarns again. I'm happy to go either way here.

Yeah, that's a good point. After this, we should make an effort to remove those NoWarns all up. @vitek-karas already did some work that will help in https://github.com/mono/linker/pull/1980.

I guess let's go with whatever is easiest at this point - keep them separate for now.